cloudhub-2/modules/ROOT/pages/ch2-deploy.adoc

```diff
@@ -46,6 +46,10 @@ CloudHub 2.0 attempts to restart the replica, using exponential backoff to avoid
 
 After a successful restart, the replica status is `STARTED` and the replica can now perform work.
 
+== Grace Period
+
+When you stop, restart, update, redeploy, or terminate an application, CloudHub 2.0 tries to delete it from Mule with a timeout period of five minutes to wait for inflight messages to be delivered. After that period, CloudHub 2.0 shuts down the whole instance. This grace period applies to the application instance, not to the runtime. The runtime uses its own configuration for the gracefully shutdown, which is, by default, five seconds.
```

Comment:
The first sentence is a little long for clarity and "for the gracefully shutdown" isn't correct grammar. With a little help from writer.ai, here's a suggestion:

When you stop, restart, update, redeploy, or terminate an application, CloudHub 2.0 tries to delete it from Mule. To make sure inflight messages are delivered, there's a five-minute timeout period. After that period, CloudHub 2.0 shuts down the whole instance. This grace period applies to the application instance, not to the runtime. The runtime uses its own configuration for shutdown. By default, the runtime grace period is five seconds.